### PR TITLE
[bitnami/phpmyadmin] Migrate ingress to networking.k8s.io/v1beta1

### PR DESCRIPTION
**Pull request by mkilchhofer**
Signed-off-by: Marco Kilchhofer <marco.kilchhofer@bedag.ch>

**Description of the change**

The phpmyadmin ingress still uses the deprecated the API group `extensions/v1beta1`.
We should use `networking.k8s.io/v1beta1` for Ingress definitions now, which is available since K8s 1.14.
https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

**Benefits**

Compatibilty with K8s 1.22+

**Possible drawbacks**

--

**Applicable issues**

--

**Additional information**

--

**Checklist** 
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [X] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by mkilchhofer** [comment]
I think the check has failed because quay.io is down:
https://status.quay.io/

File: bitnami/phpmyadmin/templates/ingress.yaml [hunk 1]
```diff
@@ -1,5 +1,9 @@
 {{- if .Values.ingress.enabled -}}
+{{- if .Capabilities.APIVersions.Has "networking.k8s.io/v1beta1" }}
+apiVersion: networking.k8s.io/v1beta1
+{{- else }}
 apiVersion: extensions/v1beta1
+{{- end }}
```
Comment (juan131) [inline bitnami/phpmyadmin/templates/ingress.yaml:6]: Can we use a macro as we did for WordPress? Check the links below:

- https://github.com/bitnami/charts/blob/master/bitnami/wordpress/templates/ingress.yaml#L2
- https://github.com/bitnami/charts/blob/master/bitnami/wordpress/templates/_helpers.tpl#L183

**Comment by mkilchhofer** [comment]
Sure, I will adapt it.

Another way would be to move this template function from wordpress to the bitnami common chart and use it from there as a subchart.

**Comment by juan131** [comment]
Correct, that's another alternative.

**Comment by mkilchhofer** [comment]
Okay. What's your favorite?
- implement it in `_helpers.tpl` of the phpmyadmin chart
- implement it in the common chart

**Comment by mkilchhofer** [comment]
Alright, I prepared everything. I need to wait for the merge of the [common-chart](https://github.com/bitnami/charts/pull/2732) then update the `requirements.lock` via the `helm dependency update` command

File: bitnami/phpmyadmin/requirements.yaml [hunk 2]
```diff
@@ -1,4 +1,9 @@
 dependencies:
+  - name: common
+    version: "^0.3.0"
```
Comment (juan131) [inline bitnami/phpmyadmin/requirements.yaml:3]: You can use:
```suggestion
    version: 0.x.x
```
I'll take the latest `0.x.x` version by default.

File: bitnami/phpmyadmin/templates/ingress.yaml [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {{- if .Values.ingress.enabled -}}
-apiVersion: extensions/v1beta1
+apiVersion: {{ include "common.capabilities.ingress.apiVersion" . }}
```
Comment (juan131) [inline bitnami/phpmyadmin/templates/ingress.yaml:2]: Nice!

File: bitnami/phpmyadmin/requirements.yaml (hunk 2 above)
Comment (mkilchhofer) [reply bitnami/phpmyadmin/requirements.yaml:3]: I can do this, but as the common chart only implement this in version 0.3.0 and higher (assumption), I used this notation to allow everything <2.0.0.

Ref: https://helm.sh/docs/topics/charts/

> Apart from version constrains employing operators `= != > < >= <=` the following shorthand notations are supported
> * hyphen ranges for closed intervals, where 1.1 - 2.3.4 is equivalent to >= 1.1 <= 2.3.4.
> * wildcards x, X and *, where 1.2.x is equivalent to >= 1.2.0 < 1.3.0.
> * tilde ranges (patch version changes allowed), where ~1.2.3 is equivalent to >= 1.2.3 < 1.3.0.
> * **caret ranges (major version changes allowed), where ^1.2.3 is equivalent to >= 1.2.3 < 2.0.0.**

Comment (juan131) [reply bitnami/phpmyadmin/requirements.yaml:3]: It's fine. Let's leave as it's.

**Pull request merged**